src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,198 @@
+package edu.hm.hafner.util;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+
+/**
+ * Test the class {@link java.util.HashSet} for integers.
+ *
+ * @author Johannes Reichle
+ */
+class SetTest {
+
+    // methods usable that are not tested yet???
+    @Test
+    void shouldCreateEmptySet() {
+        HashSet<Integer> testSet = creatHashSet(0);
+
+        Assertions.assertThat(testSet).isNotNull();
+        Assertions.assertThat(testSet).isEmpty(); // possible because AbstractSet implements Iterable
+    }
+
+    @Test
+    void shouldCreateGivenSet() {
+        ArrayList<Integer> copyList = new ArrayList<>();
+        copyList.add(0);
+        copyList.add(1);
+        copyList.add(2);
+        HashSet<Integer> testSet = new HashSet<>(copyList);
+
+        Assertions.assertThat(testSet).isNotNull();
+        Assertions.assertThat(testSet).containsExactlyInAnyOrder(0,1,2);
+    }
+
+    @Test
+    void shouldReturnIterator() {
+        HashSet<Integer> testSet = creatHashSet(3);
+        Iterator<Integer> testIterator = testSet.iterator();
+        HashSet<Integer> emptyTestSet = creatHashSet(0);
+        Iterator<Integer> emptyTestIterator = emptyTestSet.iterator();
+        HashSet<Integer> testSetWithNull = createHashSetWithOnlyNull();
+        Iterator<Integer> nullIterator = testSetWithNull.iterator();
+
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isFalse();
+
+        Assertions.assertThat(emptyTestIterator.hasNext()).isFalse();
+
+        Assertions.assertThat(nullIterator.hasNext()).isTrue();
+    }
+
+    @Test
+    void shouldReturnCorrectSize() {
+        HashSet<Integer> testSetZero = creatHashSet(0);
+        HashSet<Integer> testSetOne = creatHashSet(1);
+        HashSet<Integer> testSetTwo = creatHashSet(2);
+        HashSet<Integer> testSetHundred = creatHashSet(100);
+        HashSet<Integer> testSetWithNull = createHashSetWithNullAndSize4();
+
+        Assertions.assertThat(testSetZero.size() == 0).isTrue();
+        Assertions.assertThat(testSetOne.size() == 1).isTrue();
+        Assertions.assertThat(testSetTwo.size() == 2).isTrue();
+        Assertions.assertThat(testSetHundred.size() == 100).isTrue();
+        Assertions.assertThat(testSetWithNull.size() == 4).isTrue();
+    }
+
+    @Test
+    void shouldTestForEmptiness() {
+        HashSet<Integer> testSetEmpty = creatHashSet(0);
+        HashSet<Integer> testSetSizeOne = creatHashSet(4);
```

Comment:
Statt `4` lieber den Randwert `1` nutzen (dazu kommen wir aber später noch in der Vorlesung)